src/Models/HydrostaticFreeSurfaceModels/z_star_vertical_spacing.jl

```diff
@@ -0,0 +1,160 @@
+using Oceananigans.Grids
+using Oceananigans.ImmersedBoundaries: ZStarGridOfSomeKind
+
+#####
+##### ZStar-specific vertical spacings update
+#####
+
+# The easy case
+barotropic_velocities(free_surface::SplitExplicitFreeSurface) = free_surface.barotropic_velocities
+
+# The "harder" case, barotropic velocities are computed on the fly
+barotropic_velocities(free_surface) = nothing, nothing
+
+# Fallback 
+update_grid!(model, grid; parameters) = nothing
+
+function update_grid!(model, grid::ZStarGridOfSomeKind; parameters = :xy)
+
+    # Scalings and free surface
+    σᶜᶜ⁻  = grid.z.σᶜᶜ⁻
+    σᶜᶜⁿ  = grid.z.σᶜᶜⁿ
+    σᶠᶜⁿ  = grid.z.σᶠᶜⁿ
+    σᶜᶠⁿ  = grid.z.σᶜᶠⁿ
+    σᶠᶠⁿ  = grid.z.σᶠᶠⁿ
+    ∂t_σ  = grid.z.∂t_σ
+    ηⁿ    = grid.z.ηⁿ
+    η     = model.free_surface.η
```

Comment:
This code only works for a HydrostaticFreeSurfaceModel?

Reply:
It could make sense since it looks like it is supposed to be extended for other cases